Sources/KDTree.swift

```diff
@@ -31,7 +31,16 @@ public enum KDTree<Element: KDTreePoint> {
             let sortedValues = values.sorted { (a, b) -> Bool in
                 return a.kdDimension(currentSplittingDimension) < b.kdDimension(currentSplittingDimension)
             }
-            let median = sortedValues.count / 2
+          
+            var median = sortedValues.count / 2
+            let medianValue = sortedValues[median].kdDimension(currentSplittingDimension)
+          
+            //Ensure left subtree contains currentSplittingDimension-coordinate strictly less than its parent node
+            //Needed for 'contains' and 'removing' method.
```

Comment:
`Ensure left subtree contains currentSplittingDimension-coordinate strictly less than its parent node`
is enough reason for me :)

Future people reading this code will be fine without the `//Needed for 'contains' and 'removing' method.` comment, let's not start cluttering the code to much with too many lines of comments.